pkg/endpoint/local_ip.go

```diff
@@ -68,6 +80,65 @@ func GetLocalIPForDestination(dst string, family k8snet.IPFamily) string {
 	return localIP
 }
 
+func isValidGlobalIPv6(ip net.IP) bool {
+	if ip.To4() != nil || ip.To16() == nil {
+		return false
+	}
+	if ip.IsLoopback() || ip.IsLinkLocalUnicast() {
+		return false
+	}
+	if ip[0] == 0xfd { // fd00::/8 - avoid locally assigned ULA
+		return false
+	}
```

Comment:
The `ip.To4()` check really isn't necessary. Also we can logically combine these:

```suggestion
	if ip.To16() == nil || ip.IsLoopback() || ip.IsLinkLocalUnicast() ||
	    ip[0] == 0xfd   // fd00::/8 - avoid locally assigned ULA {
		return false
	}
```